Src/IronPythonTest/Cases/CaseExecuter.cs

```diff
@@ -169,19 +169,38 @@ class CaseExecuter {
                 proc.StartInfo.UseShellExecute = false;
                 proc.StartInfo.RedirectStandardError = proc.StartInfo.RedirectStandardOutput = testcase.Options.Redirect;
                 proc.Start();
-                var output = string.Empty;
-                if(testcase.Options.Redirect) {
-                    output = proc.StandardError.ReadToEnd();
+
+                if (testcase.Options.Redirect) {
+                    AsyncStreamReader(proc.StandardOutput, data => Console.Write(data));
+                    AsyncStreamReader(proc.StandardError, data => Console.Error.Write(data));
                 }
-                proc.WaitForExit();
-                exitCode = proc.ExitCode;
-                if(testcase.Options.Redirect && exitCode != 0) {
-                    Console.Error.WriteLine($"Error running {proc.StartInfo.FileName} {proc.StartInfo.Arguments}: ");
-                    Console.Error.WriteLine(output);
-                    Console.Error.WriteLine();
+
+                if (!proc.WaitForExit(testcase.Options.Timeout)) {
+                    proc.Kill();
+                    Console.Error.Write($"Timed out after {testcase.Options.Timeout / 1000.0} seconds.");
                 }
+                exitCode = proc.ExitCode;
```

Comment:
What will proc.ExitCode be if it's killed?

Reply:
Looks like it's -1.